networkit/cpp/graph/test/GraphToolsGTest.cpp

```diff
@@ -101,6 +101,48 @@ TEST_P(GraphToolsGTest, testDensity) {
     }
 }
 
+TEST_P(GraphToolsGTest, testVolume) {
+    constexpr count n = 100;
+    constexpr double p = 0.1;
+
+    auto doTest = [&](const Graph &G) {
+        // Volume for graph G
+        const auto volume = GraphTools::volume(G);
+
+        // Volume for either directed/undirected
+        double mod = directed() ? 1.0 : 2.0;
+        double baseG = weighted() ? static_cast<double>(G.totalEdgeWeight())
+                                  : static_cast<double>(G.numberOfEdges());
+        EXPECT_DOUBLE_EQ(volume, mod * baseG);
+ 
+        // Volume for subgraph G2
+        node seed = Aux::Random::integer(G.upperNodeIdBound());
+        auto sNodes = std::unordered_set<node>();
+
+        Graph::NodeIterator first = Graph::NodeIterator(&G, 0);
+        Graph::NodeIterator lastOfSet = Graph::NodeIterator(&G, seed);
+        Graph::NodeIterator last = Graph::NodeIterator(&G, G.upperNodeIdBound());
+
+        const auto volumeSet = GraphTools::volume(G, first, lastOfSet);
+        const auto inVolumeSet= GraphTools::inVolume(G, first, lastOfSet);
+        const auto volumeComplementSet = GraphTools::volume(G, lastOfSet, last);
+        const auto inVolumeComplementSet= GraphTools::inVolume(G, lastOfSet, last);
+
+        EXPECT_NEAR(volumeSet + volumeComplementSet, volume, 1e-7);
+        EXPECT_NEAR(inVolumeSet + inVolumeComplementSet, volume, 1e-7);
```

Comment:
I guess you can use `EXPECT_DOUBLE_EQ` here as well right?

Reply:
No, there are differences in the range of approx. 1e-5